registry/handlers/tags.go

```diff
@@ -49,6 +51,48 @@ func (th *tagsHandler) GetTags(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// do pagination if requested
+	q := r.URL.Query()
+	// get entries after latest, if any specified
+	if lastEntry := q.Get("last"); lastEntry != "" {
+		lastEntryIndex := sort.SearchStrings(tags, lastEntry)
+		if lastEntryIndex != len(tags) && tags[lastEntryIndex] == lastEntry {
+			tags = tags[lastEntryIndex+1:]
+		} else {
+			th.Errors = append(th.Errors, v2.ErrorCodePaginationLastUnknown.WithDetail(map[string]string{"last": lastEntry}))
```

Comment:
This error is the one that I wanted to double check. The spec says for `last` that  `Result set will include values lexically after last`. In this case, the search should be done, but it can still continue even if last doesn't exist but entries exist lexically after it.

@jdolitsky do you mind offering an opinion here?

Reply:
That makes perfect sense tbh. most have missed that it specified that. I sadly read it as after the index of the `last` string and values should be sorted lexically.
Seems like that should be the behavior :)

Just say the word and I will update it in this (and update the catalog PR after this is merged to do the same).

Reply:
@dmcgowan I'm here on behalf of @jdolitsky
You're correct, the spec says that  the `/v2/<name>/tags/list` endpoint should return a lexically-ordered list when the `n` query parameter is present, regardless of whether `last` is specified.

Usually the value of the `last` key would be acquired by by a `Link` header from an earlier response. Failing that, the registry should probably behave intelligently when the actual `last` value doesn't exist.

The current language in the spec is at the SHOULD level, but this will probably become a MUST by the time distribution-spec 1.0 is released.

Reply:
What I gather from this is the following. `last` does not have to match a value in the tags, but should specify where in the tags we want to get our tags from legically (so this should be changed in this PR).

I am wondering if `last` should be ignored, if no `n` is specified? Below is from the doc.

```
The client can then issue the request with the above value from the Link header, receiving the values v3 and v4. Note that n may change on the second to last response or be fully omitted, depending on the server implementation.
```

Keep in mind, this will also change the `catalog` listing to work the same way as `tags`.

Reply:
This seems related to this discussion: https://github.com/opencontainers/distribution-spec/issues/132

Reply:
Okay, so what I think should happen is the following.

- `n=0` will return a empty list, and will not include a `Link`, as there are no more entries (depends on how you look at it. But if I want 0 and I want more, I will still get 0).
- `last` will be used to find the entries lexically after it, and the index does not need to exists.
    - this could bring some weird bugs for some people regarding encoding. but it does make sense to be able to search like this.
- `last` can stand on its own without `n`, as to use the default `n` value from the server (which it may impose).

How does this sound @jdolitsky and @dmcgowan ?

Reply:
Makes sense to me, but we should probably clarify in the spec first. I'm ok making that update in this project with expectation that we may need to make slight changes if we come to a different agreement for the spec

Reply:
@dmcgowan  so this is good to go, if I make the changes and then we can update the spec in accordance?

Reply:
yeah, let's go ahead and make those changes and get this merged in. We can tweak the conformance tests and spec after, if we need to come back and sync here we can before the release

Reply:
@dmcgowan  thougt I had pinged you here, I guess not. It should be good to go! Travis is giving some weird errors, that doesn't seem to be a problem with the code. Should be ready to merge!